diff --git a/scp-dbus-service.py b/scp-dbus-service.py
index 0650cfa..ee98960 100644
--- a/scp-dbus-service.py
+++ b/scp-dbus-service.py
@@ -212,16 +212,18 @@ class GetBestDriversRequest:
                                     self.dialog.connect ('driver-download-checked',
                                                          self.on_driver_download_checked)]
 
-                    self.reply_if_fail = [(x, fit[x]) for x in ppdnamelist]
-                    if not self.dialog.init ('download_driver',
-                                             devid=self.device_id):
-                        try:
-                            g_killtimer.remove_hold ()
-                        finally:
-                            e = RuntimeError ("Failed to launch dialog")
-                            self.error_handler (r)
-
-                    return
+                    if self.dialog.init ('download_driver',
+                                         devid=self.device_id):
+                        self.loop.run()
+                    for handle in self.handles:
+                        self.dialog.disconnect (handle)
+                    if len(self.installed_files) > 0:
+                        debugprint ("GetBestDrivers request: Re-fetch PPDs after driver download")
+                        g_ppds = FetchedPPDs (self.cupsconn, self.language)
+                        self._signals.append (g_ppds.connect ('ready', self._ppds_ready))
+                        self._signals.append (g_ppds.connect ('error', self._ppds_error))
+                        g_ppds.run ()
+                        return
             except:
                 # Ignore driver download if packages needed for the GUI are not
                 # installed or if no windows can be opened
